packages/remark-cli/readme.md

```diff
@@ -86,25 +118,189 @@ Examples:
+      "remark-preset-lint-recommended", // Few recommended rules.
+      [
+        // Generate a table of contents in `## Contents`
+        "remark-toc",
+        {
+          "heading": "contents"
+        }
+      ]
+    ]
+  },
+  /* … */
+```
+
+> 👉 **Note**: you must remove the comments in the above examples when
+> copy/pasting them, as comments are not supported in `package.json` files.
+
+Finally, you can run the npm script to check and format markdown files in your
+project:
+
+```sh
+npm run format
+```
+
+### Example: config files (JSON, YAML, JS)
+
+In the previous example, we saw that `remark-cli` was configured from within a
+`package.json` file.
+That’s a good place when the configuration is relatively short, when you have a
+`package.json`, and when you don’t need comments (which are not allowed in
+JSON).
+
+You can also define configuration in separate files in different languages.
+With the `package.json` config as inspiration, here’s a JavaScript version that
+can be placed in `.remarkrc.js`:
+
+```js
+import remarkPresetLintConsistent from 'remark-preset-lint-consistent'
+import remarkPresetLintRecommended from 'remark-preset-lint-recommended'
+import remarkToc from 'remark-toc'
+
+const remarkConfig = {
+  settings: {
+    bullet: '*', // Use `*` for list item bullets (default)
+    // See <https://github.com/remarkjs/remark/tree/main/packages/remark-stringify> for more options.
+  },
+  plugins: [
+    remarkPresetLintConsistent, // Check that markdown is consistent.
+    remarkPresetLintRecommended, // Few recommended rules.
+    // Generate a table of contents in `## Contents`
+    [remarkToc, {heading: 'contents'}]
+  ]
+}
+
+export default remarkConfig
+```
+
+This is the same configuration in YAML, which can be placed in `.remarkrc.yml`:
+
+```yml
+settings:
+  bullet: "*"
+plugins:
+  # Check that markdown is consistent.
+  - remark-preset-lint-consistent
+  # Few recommended rules.
+  - remark-preset-lint-recommended
+  # Generate a table of contents in `## Contents`
+  - - remark-toc
+    - heading: contents
+```
+
+When `remark-cli` is about to process a markdown file it’ll search the file
+system upwards for configuration files starting at the folder where that file
+exists.
+Take the following file structure as an illustration:
+
+```txt
+folder/
+├─ subfolder/
+│  ├─ .remarkrc.json
+│  └─ file.md
+├─ .remarkrc.js
+├─ package.json
+└─ readme.md
+```
+
+When `folder/subfolder/file.md` is processed, the closest config file is
+`folder/subfolder/.remarkrc.json`.
+For `folder/readme.md`, it’s `folder/.remarkrc.js`.
+
+The order of precedence is as follows.
+Earlier wins (so in the above file structure `folder/.remarkrc.js` wins over
+`folder/package.json`):
+
+1.  `.remarkc` (JSON)
+2.  `.remarkc.json` (JSON)
+3.  `.remarkc.cjs` (CJS)
+4.  `.remarkc.mjs` (ESM)
+5.  `.remarkc.js` (CJS or ESM, depending on `type: 'module'` in `package.json`)
+6.  `.remarkc.yaml` (YAML)
+7.  `.remarkc.yml` (YAML)
```

Comment:
I'm confused JSON appears earlier in the list, but JS which is later in the list wins?

Reply:
Good catch. The difference is that the json in the example is called `package.json`, which is not listed here